### PR TITLE
[collect] Allow collect to upload via policy

### PR DESCRIPTION
**Pull request by TurboTurtle**
Extends the upload functionality to `sos collect`, to work the same way
that it does for `report`. Uploads will default to policy definitions,
then user provided overrides.

Resolves: #2210

Signed-off-by: Jake Hunsaker <jhunsake@redhat.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [ ] If this commit closes an existing issue, is the line `Closes: #ISSUENUMBER` included in an independent line?
- [x] If this commit resolves an existing pull request, is the line `Resolves: #PRNUMBER` included in an independent line?

### CODE REVIEW

File: sos/policies/__init__.py [hunk 1]
```diff
@@ -1402,7 +1402,7 @@ def pre_work(self):
                                            "that you are generating this "
                                            "report for [%s]: ") % caseid)
                 # Policies will need to handle the prompts for user information
-                if cmdline_opts.upload or self.upload_url:
+                if cmdline_opts.upload and self.get_upload_url():
```
Comment (pmoravec) [inline sos/policies/__init__.py:1405]: Why this change (esp. compared to "upload or upload_url" at https://github.com/sosreport/sos/pull/2210/files#diff-e07b4b5df8f32d4dab336b60233ae2aeR1130), please?

Otherwise nice extension.

Comment (TurboTurtle) [reply sos/policies/__init__.py:1405]: Hmm, the prompts should set the necessary bits to allow the latter addition to execute appropriately - but from a consistency standpoint you're absolutely right. I'll change the latter to also check `get_upload_url()`.

**Review by pmoravec — APPROVED** [review]
